### PR TITLE
Add Worker Factory Comments

### PR DESCRIPTION
**Pull request by halakaraki**

### CODE REVIEW

File: src/main/java/com/uber/cadence/worker/Worker.java [hunk 1]
```diff
@@ -423,10 +467,29 @@ public Factory(IWorkflowService workflowService, String domain, FactoryOptions f
               metricsScope);
     }
 
+    /**
+     * Creates worker that connects to an instance of the Cadence Service. It uses the domain *
```
Comment (mfateev) [inline src/main/java/com/uber/cadence/worker/Worker.java]: reformat the doc, remove trailing *

File: src/main/java/com/uber/cadence/worker/Worker.java [hunk 2]
```diff
@@ -423,10 +467,29 @@ public Factory(IWorkflowService workflowService, String domain, FactoryOptions f
...
+     * configured at the Factory level. New workers cannot be created after the start() has been
+     * called
+     *
+     * @param taskList task list name worker uses to poll. It uses this name for both decision and *
```
Comment (mfateev) [inline src/main/java/com/uber/cadence/worker/Worker.java]: trailing *

File: src/main/java/com/uber/cadence/worker/Worker.java [hunk 3]
```diff
@@ -528,28 +597,49 @@ private String getStickyTaskListName() {
       private PollerOptions stickyWorkflowPollerOptions;
       private Scope metricScope;
 
+      /**
+       * When set to true it will create an affinity between the worker and the workflow run it's
+       * processing. Workers will cache workflows and will handle all decisions for that workflow
+       * instance until it's complete or evicted from the cache.
+       */
       public Builder setEnableStickyExecution(boolean enableStickyExecution) {
```
Comment (mfateev) [inline src/main/java/com/uber/cadence/worker/Worker.java]: Let's change it to setDisableStickyExecution as the default value of a boolean property is false and we want sticky on by default.

File: src/main/java/com/uber/cadence/worker/Worker.java [hunk 4]
```diff
@@ -528,28 +597,49 @@ private String getStickyTaskListName() {
...
         this.enableStickyExecution = enableStickyExecution;
         return this;
       }
 
+      /**
+       * When Sticky execution is enabled this will set the maximum allowed number of workflows
+       * cached. This cache is shared by all workers created by the Factory.
```
Comment (mfateev) [inline src/main/java/com/uber/cadence/worker/Worker.java]: Document the default value

File: src/main/java/com/uber/cadence/worker/Worker.java [hunk 5]
```diff
@@ -528,28 +597,49 @@ private String getStickyTaskListName() {
...
+       */
       public Builder setCacheMaximumSize(int cacheMaximumSize) {
         this.cacheMaximumSize = cacheMaximumSize;
         return this;
       }
 
+      /**
+       * Maximum number of threads available for workflow execution across all workers created by
+       * the Factory.
+       */
```
Comment (mfateev) [inline src/main/java/com/uber/cadence/worker/Worker.java]: Document the default value.
Rename to set**M**axWorkflowThreadCount

File: src/main/java/com/uber/cadence/worker/Worker.java [hunk 6]
```diff
@@ -528,28 +597,49 @@ private String getStickyTaskListName() {
...
       public Builder setmaxWorkflowThreadCount(int maxWorkflowThreadCount) {
         this.maxWorkflowThreadCount = maxWorkflowThreadCount;
         return this;
       }
 
+      /**
+       * Timeout for sticky workflow decision to be picked up by the host assigned to it. Once it
+       * times out then it can be picked up by any worker.
+       */
```
Comment (mfateev) [inline src/main/java/com/uber/cadence/worker/Worker.java]: Document the default value.

File: src/main/java/com/uber/cadence/worker/Worker.java (hunk 4 above)
Comment (halakaraki) [reply src/main/java/com/uber/cadence/worker/Worker.java]: resolved

File: src/main/java/com/uber/cadence/worker/Worker.java (hunk 5 above)
Comment (halakaraki) [reply src/main/java/com/uber/cadence/worker/Worker.java]: resolved

File: src/main/java/com/uber/cadence/worker/Worker.java (hunk 6 above)
Comment (halakaraki) [reply src/main/java/com/uber/cadence/worker/Worker.java]: resolved

File: src/main/java/com/uber/cadence/worker/Worker.java (hunk 2 above)
Comment (halakaraki) [reply src/main/java/com/uber/cadence/worker/Worker.java]: resolved

File: src/main/java/com/uber/cadence/worker/Worker.java (hunk 1 above)
Comment (halakaraki) [reply src/main/java/com/uber/cadence/worker/Worker.java]: resolved

File: src/main/java/com/uber/cadence/worker/Worker.java (hunk 3 above)
Comment (halakaraki) [reply src/main/java/com/uber/cadence/worker/Worker.java]: Will do this in a separate CR before release

**Pull request closed without merging**